pageserver/src/bin/pageserver.rs

```diff
@@ -147,17 +143,35 @@ fn main() -> anyhow::Result<()> {
 }
 
 fn initialize_config(
+    identity_file_path: &Utf8Path,
     cfg_file_path: &Utf8Path,
     arg_matches: clap::ArgMatches,
     workdir: &Utf8Path,
-) -> anyhow::Result<ControlFlow<(), &'static PageServerConf>> {
-    let init = arg_matches.get_flag("init");
+) -> anyhow::Result<&'static PageServerConf> {
+    // An identity file may be present on disk for new pageservers.
+    // If that's the case, use it as the source of truth for the node id.
+    // Otherwise, use the id from the pageserver config file.
+    let identity = match std::fs::File::open(identity_file_path) {
```

Comment:
Falling back to the `pageserver.toml` isn't necessary / is unneeded complexity because the `pageserver.toml` written by https://github.com/neondatabase/aws/pull/1322 will not contain the `id` key.